lib/charms/opensearch/v0/helper_databag.py

```diff
@@ -4,9 +4,13 @@
 """Utility classes for app / unit data bag related operations."""
 
 import json
-from typing import Dict, Optional
+from abc import ABC, abstractmethod
+from ast import literal_eval
+from typing import Dict, Optional, Union
 
+from charms.opensearch.v0.constants_tls import CertType
 from charms.opensearch.v0.helper_enums import BaseStrEnum
```

Comment:
Hmm, the reason I'm using my own `BaseStrEnum` here, is due to the default string representation of an enum value and me wanting to avoid to having to use a suffix `.value` in an interpolated string containing the enum value.
So, `print(Enum.A)` would print `Enum.A`, in order to get the value I'd have to add `.value` which makes strings longer, but `print(BaseStrEnum.A)` would print `A_Value`.

The use case is similar to the link you had shared about making an enum extend `str`. It's a temporary definition while the our CI workers come with `python 3.11`, then I would get rid of it in favor of [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum)